core/domain/acl_decorators.py

```diff
@@ -1033,3 +1056,41 @@ def test_can_manage_topic_rights(self, topic_id, **kwargs):
     test_can_manage_topic_rights.__wrapped__ = True
 
     return test_can_manage_topic_rights
+
+
+def can_accept_suggestion(handler):
+    """Decorator to check if user can accept the given suggestion."""
+    def test_can_accept(self, suggestion_id, **kwargs):
+        """Checks if the user can accept the suggestion.
+
+        Args:
+            suggestion_id: str. The suggestion id.
+            **kwargs: *. Keyword arguments.
+
+        Returns:
+            *. The return value of the decorated function.
+        """
+        if not self.user_id:
+            raise base.UserFacingExceptions.NotLoggedInException
+
+        print self.user
+        if len(suggestion_id.split('.')) != 3:
+            raise self.InvalidInputException('Invalid format for suggestion_id.'
+                                             ' It must contain 3 parts'
+                                             ' separated by \'.\'')
+        target_type = suggestion_id.split('.')[0]
+
+        if target_type == suggestion_models.TARGET_TYPE_EXPLORATION:
+            exploration_id = suggestion_id.split('.')[1]
+            exploration_rights = rights_manager.get_exploration_rights(
+                exploration_id)
+            if rights_manager.check_can_edit_activity(
```

Comment:
In the future, this has to be updated to account for the user scores too, right? Maybe worth putting a TODO here ....

Reply:
Yep, the checks for user scores will go into this decorator. Will add a TODO!

Reply:
Done